snakemake/io.py

```diff
@@ -68,11 +65,11 @@ class AnnotatedStringInterface(ABC):
     @property
     @abstractmethod
     def flags(self) -> Dict[str, Any]:
-        ...
+        return NotImplemented
```

Comment:
Why this change? Using the ellipsis literal for abstract methods is common practice. Is it because of the type hint?

Reply:
I commented on the PR to explain this, but for some reason the CI version of black insists the ellipsis belongs on a separate line, and my version of black insists it belongs at the end of the line, and I couldn't find why this keeps happening.

Reply:
Ahh, I see, this is related to changes in black version 24+, they changed their opinions on several fronts. CI uses black 23.12, I use 24.4. I will pin it to 23.12 in my dev environment and revert this if that fixes it.